pytential/symbolic/pde/beltrami.py

```diff
@@ -0,0 +1,392 @@
+
+    def prepare_rhs(self, b: sym.var) -> sym.var:
+        """
+        :returns: a modified expression for the right-hands ide *b* based on
+            the preconditiong used in the operator.
+        """
+        S = partial(sym.S, self.kernel,     # noqa: N806
+                qbx_forced_limit=+1, **self.kernel_arguments)
+
+        if self.precond == "left":
+            return S(b)
+        else:
+            return b
+
+    def _get_left_operator(self, sigma: sym.var, kappa: sym.var, **kwargs: Any):
+        from sumpy.kernel import LaplaceKernel
+        if isinstance(self.kernel, LaplaceKernel):
+            raise TypeError(
+                    f"{type(self).__name__} does not support the Laplace "
+                    "kernel, use LaplaceBeltramiOperator instead")
+
+        context = self.kernel_arguments.copy()
+        context.update(kwargs)
+
+        knl = self.kernel
+        lknl = LaplaceKernel(knl.dim)
+
+        # {{{ layer potentials
+
+        # laplace
+        S0 = partial(sym.S, lknl, qbx_forced_limit=+1, **kwargs)        # noqa: N806
+        D0 = partial(sym.D, lknl, qbx_forced_limit="avg", **kwargs)     # noqa: N806
+        Dp0 = partial(sym.Dp, lknl, qbx_forced_limit="avg", **kwargs)   # noqa: N806
+
+        # base
+        S = partial(sym.S, knl, qbx_forced_limit=+1, **context)         # noqa: N806
+        Sp = partial(sym.Sp, knl, qbx_forced_limit="avg", **context)    # noqa: N806
+        Spp = partial(sym.Spp, knl, qbx_forced_limit="avg", **context)  # noqa: N806
+
+        # }}}
+
+        # {{{ operator
+
+        # similar to 6.2 in [ONeil2017]
+        op = (
+                sigma / 4
+                - D0(D0(sigma))
+                + S(kappa * Sp(sigma))
+                + S(Spp(sigma) + Dp0(sigma))
+                - S(Dp0(sigma))
+                + S0(Dp0(sigma))
+                )
+
+        # }}}
+
+        return op
+
+    def _get_right_operator(self, sigma: sym.var, kappa: sym.var, **kwargs: Any):
+        from sumpy.kernel import LaplaceKernel
+        if isinstance(self.kernel, LaplaceKernel):
+            raise TypeError(
+                    f"{type(self).__name__} does not support the Laplace "
+                    "kernel, use LaplaceBeltramiOperator instead")
+
+        context = self.kernel_arguments.copy()
+        context.update(kwargs)
+
+        knl = self.kernel
+        lknl = LaplaceKernel(knl.dim)
+
+        # {{{ layer potentials
+
+        # laplace
+        S0 = partial(sym.S, lknl, qbx_forced_limit=+1, **kwargs)        # noqa: N806
+        Sp0 = partial(sym.Sp, lknl, qbx_forced_limit="avg", **kwargs)   # noqa: N806
+        Dp0 = partial(sym.Dp, lknl, qbx_forced_limit="avg", **kwargs)   # noqa: N806
+
+        # base
+        S = partial(sym.S, knl, qbx_forced_limit=+1, **context)         # noqa: N806
+        Sp = partial(sym.Sp, knl, qbx_forced_limit="avg", **context)    # noqa: N806
+        Spp = partial(sym.Spp, knl, qbx_forced_limit="avg", **context)  # noqa: N806
```

Comment:
Avoid duplicating this code between above and here?

Reply:
Went ahead and removed those `_get_left_operator` and `_get_right_operator` and unified this duplicated code in f922ecf.